eth/client.go

```diff
@@ -883,84 +888,31 @@ func (c *client) ContractAddresses() map[string]ethcommon.Address {
 }
 
 func (c *client) CheckTx(tx *types.Transaction) error {
-	ctx, cancel := context.WithTimeout(context.Background(), c.txTimeout)
-	defer cancel()
-
-	receipt, err := bind.WaitMined(ctx, c.backend, tx)
-	if err != nil {
-		return err
+	txs := make(chan *types.Receipt, 10)
+	b, ok := c.backend.(*backend)
+	if !ok {
+		return fmt.Errorf("error retrieving client")
+	}
+	txSub := b.tm.Subscribe(txs)
+	defer txSub.Unsubscribe()
+
+txLoop:
+	for {
+		select {
+		case err := <-txSub.Err():
+			return err
+		case event := <-txs:
+			if tx.Hash() == event.TxHash {
```

Comment:
Looks like this TransactionManager API currently implies that a tx will always be confirmed which is signaled by receiving a receipt in the subscription channel. Do we really want the API to make this guarantee to the caller? If we end up adding a max # of replacements there seems like there could be edge cases where the max # of replacements is hit without the tx being confirmed. In that scenario, a receipt would never be received in the subscription channel and the caller would wait forever. Perhaps the API should give the caller a receipt *or* an error?

Reply:
Yeah in retrospect I like your idea better than the current implementation, this will also give us access to revert reason error messages being returned back through `checkTx` 

Updated this API and its implementation in 41d57f11cc1d93e63a2345f082783d00eb600e24